TACC/Core-Portal
# task/FP-618, task/FP-588: Adjustments to marking notifications as "read"

## Overview: ##
- Mark notifications as read when visiting the history page
- Disable or hide "Mark All as Viewed" if no notifications are available

## PR Status: ##


## Related Jira tickets: ##

* [FP-618](https://jira.tacc.utexas.edu/browse/FP-618)
* [FP-588](https://jira.tacc.utexas.edu/browse/FP-588)

## Summary of Changes: ##
- Adjusts response 

## Testing Steps: ##
0. Run an [ngrok](https://ngrok.com/download) session to route webhooks to `frontera_prtl_nginx`, and update `_WH_BASE_URL`:
```
ngrok http 443
```
1. Launch a job
2. Go to https://cep.dev/workbench/history/jobs
3. Confirm that the badge counter resets to 0, while maintaining the highlighting of the updated job row
4. Either visit the tab again or click "Mark All as Read" to remove purple highlighting
5. Confirm "Mark All as Read" button is not visible when no notifications are unread

## UI Photos:
<img width="1672" alt="Screen Shot 2020-08-18 at 11 43 13 PM" src="https://user-images.githubusercontent.com/20326896/90593073-af4e7300-e1ac-11ea-97a2-d47a5f58b60c.png">


## Notes: ##
- `HttpResponse` was causing `fetchUtil` to error out when generating the response, since `response.json()` fails.
    - Future TODO: Update all django api endpoints to respond with `JsonResponse`

Outstanding tasks (future PR?):
- [ ] Auto mark as read for new notifs in a subview, but keep the purple highlighting
- [ ] Remove slight delay of removing purple highlighting of already read notifs when coming back to a notif view (probably via a spinner, or a `useHistory` check to see if referring url is a history subview)
